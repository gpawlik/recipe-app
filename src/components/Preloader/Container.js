import styled from 'styled-components';

const Container = styled.div`
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background-color: rgba(255, 255, 255, 0.4);
  display: flex;
  align-items: center;
  justify-content: center;
  z-index: 99;
`;

export default Container;
