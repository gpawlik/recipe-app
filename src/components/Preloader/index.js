import React from 'react';

import Spinner from 'components/Spinner';
import Container from './Container';

const Preloader = () => {
  return (
    <Container>
      <Spinner />
    </Container>
  );
};

export default Preloader;
