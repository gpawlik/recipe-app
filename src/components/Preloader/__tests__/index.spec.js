import React from 'react';
import { create } from 'react-test-renderer';

import Preloader from '../';

const initialProps = {};

const mockComponent = props => {
  const finalProps = {
    ...initialProps,
    ...props
  };

  return (
    <Preloader {...finalProps} />
  );
};

describe('Preloader', () => {
  it('should render with the default props', () => {
    const tree = create(mockComponent()).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
