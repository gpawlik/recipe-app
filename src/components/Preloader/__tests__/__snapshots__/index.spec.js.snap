// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Preloader should render with the default props 1`] = `
<div
  className="sc-htpNat cQsnYc"
>
  <svg
    className="sc-bwzfXH bEKcUv"
    height={66}
    viewBox="0 0 66 66"
    width={66}
  >
    <circle
      className="sc-bdVaJa eiyToN"
      cx={33}
      cy={33}
      fill="none"
      r="30"
    />
  </svg>
</div>
`;
